control-plane/pkg/reconciler/kafka/kafka.go

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright 2022 The Knative Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kafka
+
+import (
+	"fmt"
+	"strconv"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kedav1alpha1 "knative.dev/eventing-autoscaler-keda/third_party/pkg/apis/keda/v1alpha1"
+	"knative.dev/pkg/kmeta"
+	"knative.dev/pkg/reconciler"
+
+	"knative.dev/eventing-autoscaler-keda/pkg/reconciler/keda"
+	kafkainternals "knative.dev/eventing-kafka-broker/control-plane/pkg/apis/internals/kafka/eventing/v1alpha1"
+)
+
+const (
+	defaultKafkaLagThreshold = 10
+)
+
+func GenerateScaleTarget(cg *kafkainternals.ConsumerGroup) *kedav1alpha1.ScaleTarget {
+	return &kedav1alpha1.ScaleTarget{
+		Name:       cg.Name,
+		APIVersion: kafkainternals.SchemeGroupVersion.String(),
+		Kind:       "ConsumerGroup",
+	}
+}
+
+func GenerateScaleTriggers(cg *kafkainternals.ConsumerGroup, triggerAuthentication *kedav1alpha1.TriggerAuthentication) ([]kedav1alpha1.ScaleTriggers, error) {
+	triggers := []kedav1alpha1.ScaleTriggers{}
+	bootstrapServers := cg.Spec.Template.Spec.Configs.Configs["bootstrap.servers"]
+	consumerGroup := cg.Spec.Template.Spec.Configs.Configs["group.id"]
+
+	lagThreshold, err := keda.GetInt32ValueFromMap(cg.Annotations, keda.KedaAutoscalingKafkaLagThreshold, defaultKafkaLagThreshold)
+	if err != nil {
+		return nil, err
+	}
+
+	allowIdleConsumers := "false"
+	if cg.Status.Placements != nil {
+		allowIdleConsumers = "true"
+	}
+
+	for _, topic := range cg.Spec.Template.Spec.Topics {
+		triggerMetadata := map[string]string{
+			"bootstrapServers":   bootstrapServers,
+			"consumerGroup":      consumerGroup,
+			"topic":              topic,
+			"lagThreshold":       strconv.Itoa(int(*lagThreshold)),
+			"allowIdleConsumers": allowIdleConsumers,
+		}
+
+		trigger := kedav1alpha1.ScaleTriggers{
+			Type:              "kafka",
+			Metadata:          triggerMetadata,
+			AuthenticationRef: &kedav1alpha1.ScaledObjectAuthRef{},
+		}
+
+		if triggerAuthentication != nil {
+			trigger.AuthenticationRef.Name = triggerAuthentication.Name
+		}
+
+		triggers = append(triggers, trigger)
+	}
+
+	return triggers, nil
+}
+
+func GenerateTriggerAuthentication(cg *kafkainternals.ConsumerGroup, saslType *string) (*kedav1alpha1.TriggerAuthentication, *corev1.Secret, error) {
+
+	secretTargetRefs := []kedav1alpha1.AuthSecretTargetRef{}
```

Comment:
nit: minor optimization for allocations:
```suggestion
	secretTargetRefs := make([]kedav1alpha1.AuthSecretTargetRef, 0, 8)
```